packages/user-extension/src/index.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+/**
+ * @packageDocumentation
+ * @module user-extension
+ */
+
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+import { IStateDB } from '@jupyterlab/statedb';
+import {
+  ICurrentUser,
+  IUserMenu,
+  RendererUserMenu,
+  User,
+  UserMenu
+} from '@jupyterlab/user';
+import { Menu, MenuBar, Widget } from '@lumino/widgets';
+
+/**
+ * A namespace for command IDs.
+ */
+export namespace CommandIDs {
+  export const settings = '@jupyterlab/user-extension:settings:open';
+}
+
+/**
+ *
+ */
+const userPlugin: JupyterFrontEndPlugin<User> = {
+  id: '@jupyterlab/user-extension:user',
+  autoStart: true,
+  requires: [IStateDB],
+  provides: ICurrentUser,
+  activate: (app: JupyterFrontEnd, state: IStateDB): User => {
+    return new User(state);
+  }
+};
+
+/**
+ *
+ */
+const userMenuPlugin: JupyterFrontEndPlugin<Menu> = {
+  id: '@jupyterlab/user-extension:userMenu',
+  autoStart: true,
+  requires: [ICurrentUser],
+  provides: IUserMenu,
+  activate: (app: JupyterFrontEnd, user: User): Menu => {
+    const { shell, commands } = app;
+
+    const spacer = new Widget();
+    spacer.id = 'jp-topbar-spacer';
+    shell.add(spacer, 'top', { rank: 1000 });
```

Comment:
This is gonna be a problem for the two following reasons:

- Not enough rank between 1000 and 1002: if user wants to place more element on that top bar align right but keeping the user at the far right we should offer more empty slots
- If the plugin is not loaded then the positioning of a third party extension will change if it wants it to be rightly aligned depending if this plugin is loaded or not. So I would always add the spacer in the top bar.

Reply:
May the menubar should be 2000?

Reply:
Currently the elements added to the top bar have rank 100. The default rank is 900. So i proposed to set the spacer at the default rank of 900 and to set the new menubar at 1000.